src/send_kcidb.py

```diff
@@ -486,12 +488,61 @@ def _get_test_data_recursively(self, node, origin, parsed_test_node, parsed_buil
                 self._get_test_data_recursively(child, origin, parsed_test_node,
                                                 parsed_build_node)
 
+    def _node_processed(self, node):
+        """
+        Mark the node as processed, sent_kcidb field to True
+        That means kcidb has received the node, and at least tried to process it
+        Data in node might be invalid, and not really sent to kcidb
+        This is workaround, until we improve event handling in kernelci-pipeline/api
+        """
+        node['processed_by_kcidb_bridge'] = True
+        try:
+            self._api.node.update(node)
+        except Exception as exc:
+            self.log.error(f"Failed to update node {node['id']}: {str(exc)}")
+
+    def _node_processed_recursively(self, node):
+        """
+        Mark the node and its child nodes as processed
+        """
+        child_nodes = self._api.node.find({'parent': node['id']})
+        if child_nodes:
+            for child in child_nodes:
+                self._node_processed_recursively(child)
```

Comment:
Instead of querying API, we should be getting node IDs from `revision` data as there might be some cases when not all the child nodes are submitted to KCIDB.
So, mark `sent_kcidb` to `True` only if the node was actually submitted.

Reply:
At the moment we mark just "processed by kcidb bridge", to workaround missing events issue.

Reply:
Okay. So, `sent_to_kcidb` would be a different flag for successful submission. Got it.